File: frontend/apps/android/app/src/main/java/app/pantopus/android/data/api/models/scheduling/InvoiceDtos.kt
@file:Suppress("PackageNaming")

package app.pantopus.android.data.api.models.scheduling

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Business invoices (gig-system reuse). `GET /invoices`, `GET /invoices/:id`,
 * `POST /invoices/:id/send` (business-only). `line_items` is a flexible array.
 */
@JsonClass(generateAdapter = true)
data class InvoiceDto(
    val id: String,
    @Json(name = "business_user_id") val businessUserId: String? = null,
    @Json(name = "recipient_user_id") val recipientUserId: String? = null,
    @Json(name = "total_cents") val totalCents: Int? = null,
    @Json(name = "subtotal_cents") val subtotalCents: Int? = null,
    @Json(name = "fee_cents") val feeCents: Int? = null,
    val currency: String? = null,
    /** BusinessInvoice lifecycle: draft / sent / viewed / paid / void / overdue. */
    val status: String? = null,
    @Json(name = "due_date") val dueDate: String? = null,
    @Json(name = "paid_at") val paidAt: String? = null,
    @Json(name = "line_items") val lineItems: List<Map<String, Any?>>? = null,
    @Json(name = "created_at") val createdAt: String? = null,
)

/** `GET /invoices` — `{ invoices: [...] }`. */
@JsonClass(generateAdapter = true)
data class GetInvoicesResponse(
    val invoices: List<InvoiceDto> = emptyList(),
)

/** `GET /invoices/:id` — `{ invoice: … }`. */
@JsonClass(generateAdapter = true)
data class InvoiceResponse(
    val invoice: InvoiceDto,
)

File: frontend/apps/ios/Pantopus/Core/Networking/Models/Scheduling/InvoiceDTOs.swift
//
//  InvoiceDTOs.swift
//  Pantopus
//
//  DTOs for business scheduling invoices — routes `/api/scheduling/invoices*`
//  (business-only; empty for non-business owners). See
//  `reference/calendarly-backend-api.md`.
//

import Foundation

/// A business invoice (gig-system reuse).
public struct InvoiceDTO: Decodable, Sendable, Hashable, Identifiable {
    public let id: String
    public let businessUserId: String?
    public let recipientUserId: String?
    public let totalCents: Int?
    public let currency: String?
    /// Structured line items (shape varies); kept as flexible JSON.
    public let lineItems: JSONValue?
    public let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case businessUserId = "business_user_id"
        case recipientUserId = "recipient_user_id"
        case totalCents = "total_cents"
        case currency
        case lineItems = "line_items"
        case createdAt = "created_at"
    }
}

/// `GET /invoices` → `{ invoices }`.
public struct InvoicesResponse: Decodable, Sendable, Hashable {
    public let invoices: [InvoiceDTO]
}

/// `GET /invoices/:id` → `{ invoice }`.
public struct InvoiceResponse: Decodable, Sendable, Hashable {
    public let invoice: InvoiceDTO
}
